iohk-monitoring/src/Cardano/BM/Output/Monitoring.lhs

```diff
@@ -149,8 +149,11 @@ spawnDispatcher mqueue config sbtrace = do
 Inspect the log message and match it against configured thresholds. If positive,
 then run the action on the current state and return the updated state.
 \begin{code}
-evalMonitoringAction :: MonitorMap -> LogObject a -> IO MonitorMap
-evalMonitoringAction mmap logObj@(LogObject logname _ _) =
+evalMonitoringAction :: Trace.Trace IO a
+                     -> MonitorMap
+                     -> LogObject a
+                     -> IO MonitorMap
+evalMonitoringAction sbtrace mmap logObj@(LogObject logname _ _) =
     case HM.lookup logname mmap of
```

Comment:
it just needs to have the name of LogObject added to the `sbtrace`, otherwise we can't distinguish between monitors..

Reply:
Should it look like this?

```
evalMonitoringAction :: Trace.Trace IO a
                     -> MonitorMap
                     -> LogObject a
                     -> IO MonitorMap
evalMonitoringAction sbtrace mmap logObj@(LogObject logname _ _) = do
    sbtraceWithLogName <- Trace.appendName logname sbtrace
    case HM.lookup sbtraceWithLogName mmap of
    ...
```